openhexa/toolbox/dhis2/dhis2.py

```diff
@@ -51,24 +51,23 @@ def identifiable_objects(self, uid: str) -> dict:
         r = self.client.api.get(f"identifiableObjects/{uid}")
         return r
 
-    def organisation_unit_levels(self) -> List[dict]:
+    def organisation_unit_levels(self, fields: str = "id,name,level") -> List[dict]:
         """Get names of all organisation unit levels.
 
+        Parameters
+        ----------
+        fields: str, optional
+            DHIS2 fields to include in the response, where default value is "id,name,level"
         Return
         ------
         list of dict
             Id, number and name of each org unit level.
         """
-        r = self.client.api.get("filledOrganisationUnitLevels")
+        params = {"fields": fields}
+        r = self.client.api.get("filledOrganisationUnitLevels", params=params)
         levels = []
         for level in r:
-            levels.append(
-                {
-                    "id": level.get("id"),
-                    "level": level.get("level"),
-                    "name": level.get("name"),
-                }
-            )
+            levels.append({k: v for k, v in level.items() if k in fields.split(",")})
```

Comment:
[Minor] `fields.split(",")` can be extracted from the loop, it improves a bit the performance